src/main/java/org/mariadb/jdbc/internal/stream/PacketOutputStream.java

```diff
@@ -450,7 +414,6 @@ public PacketOutputStream writeByteArray(final byte[] bytes) {
      * @return this.
      */
     public PacketOutputStream writeByteArrayLength(final byte[] bytes) {
-        assureBufferCapacity(bytes.length + 9);
```

Comment:
It seems like in some cases this assure capacity could be a performance saver.  If the bytes are larger than your default buffer, and if there is not enough room to write the length, you would allocate for the length write, and then allocate again for the bytes.